### DISCUSSION TITLE
add ./mvn/ config files support

### DISCUSSION THREAD

**Discussion by Cramer-0xbit** [redhat-developer/vscode-java#3096] [Ideas]
greetings

 I usually love to use vscode with java, 
 
 but when  you are in a ofuscated enviroment  like a bank  there's a lot of manual  configuration  for each project
 
 (vpn connections, proxies behind proxies  and all that  stuff)
 
 hopefully  I resolved that in maven with ```.mvn/maven.config```   [maven configuration files](https://maven.apache.org/configure.html)
 
 but when I launch vscode,  it gets mad with the inner building task,
  even if launch inside  ```maven compile``` with no errors it fails  with the    ```java (build)``` task and i dont know how to add vmArgs settings in the build task
 (i know in launch.json and settings.json but both doesn't solve the problem)
 
 but even if I solve  i think it could be great to the build task to add support to the maven configuration files to
 make the config more out of the box
 
 ![imagen](https://user-images.githubusercontent.com/12005786/236191101-fcbb8933-ef24-4d6e-84db-05b0e6549b55.png)
thanks for reading

**Comment by snjeza** [comment]
> ...  i dont know how to add vmArgs settings in the build task

You can try to add
```
 "java.jdt.ls.vmargs": "-Dmaven.resolver.transport=wagon <your_other_properties>",
```
to your settings.json

**Reply by Cramer-0xbit** [reply]
already tried but i doesnt work

**Comment by snjeza** [comment]
A related issue - https://github.com/eclipse-m2e/m2e-core/issues/274

**Comment by snjeza** [comment]
> already tried but i doesnt work

Could you attach a project example, user and workspace settings.json?

**Comment by Cramer-0xbit** [comment]
### workspace settings
```json
{
    "java.compile.nullAnalysis.mode": "automatic",
    "java.configuration.updateBuildConfiguration": "automatic",
    "java.jdt.ls.vmargs": "-Dmaven.resolver.transport=wagon -Dhttp.proxyHost=10.126.132.142 -Dhttp.proxyPort=3128 -Dhttps.proxyHost=10.126.132.142 -Dhttps.proxyPort=3128 -Djavax.net.ssl.trustStoreType=Windows-ROOT -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx2G -Xms100m -Xlog:disable",
    "java.debug.settings.vmArgs": "-Dmaven.resolver.transport=wagon -Dhttp.proxyHost=10.126.132.142 -Dhttp.proxyPort=3128 -Dhttps.proxyHost=10.126.132.142 -Dhttps.proxyPort=3128 -Djavax.net.ssl.trustStoreType=Windows-ROOT -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx2G -Xms100m -Xlog:disable",
}
```
### user settings
``` json
{
    "terminal.integrated.fontFamily": "CaskaydiaCove NFM",
    "terminal.integrated.fontWeight": "normal",
    "terminal.integrated.fontWeightBold": "bold",
    "editor.formatOnPaste": true,
    "http.systemCertificates": true,
    "gitlab.ignoreCertificateErrors": true,
    "java.configuration.maven.userSettings": "C:/Users/scarlonr/.m2/settings.xml",

    "explorer.copyRelativePathSeparator": "/",
    "java.configuration.runtimes": [
        {
            "default": true,
            "name": "JavaSE-17",
            "path": "C:/Program Files/Microsoft/jdk-17.0.7.7-hotspot",
            "sources": "C:/Program Files/Microsoft/jdk-17.0.7.7-hotspot/lib/src.zip"
        },
        {
            "name": "JavaSE-11",
            "path": "C:/Program Files/Microsoft/jdk-11.0.19.7-hotspot",
            "sources": "C:/Program Files/Microsoft/jdk-11.0.19.7-hotspot/lib/src.zip"
        },
        {
            "name": "JavaSE-1.8",
            "path": "C:/Program Files/Eclipse Adoptium/jdk-8.0.362.9-hotspot",
            "sources": "C:/Program Files/Eclipse Adoptium/jdk-8.0.362.9-hotspot/src.zip"
        }
    ],
    "grunt.autoDetect": "on",
    "gulp.autoDetect": "on",
    "jake.autoDetect": "on",
    "redhat.telemetry.enabled": true,
    "git.autofetch": true,
    "git.confirmSync": false,
    "apicurio.http.port": 8093,
    "apicurio.tools.preview.OPENAPI": false,
    "apicurio.http.host": "10.126.132.142",
    "apicurio.http.path": "/",
    "apicurio.tools.preview.format": false,
    "http.proxySupport": "fallback",
    "openapi.defaultPreviewRenderer": "swaggerui",
    "openapi.approvedHostnames": [
        "contractserver-micro-server-1.pro.int.srv.blablabank.com"
    ],
    "yaml.maxItemsComputed": 10000,
    "sonarlint.connectedMode.connections.sonarqube": [
        {
            "connectionId": "blablabla",
            "serverUrl": "https://sonar-a3mson-sonars.pro.ap.intranet.cloud.blablabla.es",
            "token": "9502a588f7676fcaf6ae660e6df9c2fb409554a3"
        }
    ],
    "diffEditor.ignoreTrimWhitespace": false,
    "[java]": {
        "editor.defaultFormatter": "redhat.java"
    },
    "plantuml.jar": "C:\\Users\\scarlonr\\herramientas\\plantuml-1.2023.4.jar",
    "boot-java.rewrite.reconcile": true,
    "cSpell.language": "en,es-ES,es",
    "workbench.iconTheme": "material-icon-theme"
}
```

**Comment by snjeza** [comment]
@Cramer-0xbit Could you show your pom.xml?

**Comment by Cramer-0xbit** [comment]
sorry for the late reply
 i dont think that showing the maven pom points any  solution, plus it has a lot of confidential data
my openapi Contract  has a losts refs of pointing  to  contract server
```json
 $ref:            https://contractserver-micro-server-1.pro.int.srv.famousbusiness.com
```
 wich is behind a proxy and maven resolves it well with my config,
But the problem of  voscode-java-plugin is that it ignores the build created by maven and tries to rebuild all itself ( included the open api generator)

 I need any way to add to the BUILD TASK the necesary proxies,
 NOT the settings.json NOT the launch.json i mean the task.son

**Reply by Cramer-0xbit** [reply]
or even better do a only compile task  when building like intelij does, keeping maven as  necessary step to build

if is a contract intelj does not regenerate  it, it waits maven to do it for itself

**Comment by snjeza** [comment]
@Cramer-0xbit Could you try to add

```
"http.proxy": "http://10.126.132.142:3128",
"http.proxySupport": "override",
```
or
```
"http.proxy": "https://10.126.132.142:3128",
"http.proxySupport": "override",
```

**Reply by Cramer-0xbit** [reply]
tried with 2 proxies neither work same error on mojo  open api generator,  but this was not the best soulution because it  works  for connecting all the editor to  the proxy  and I  need only a minimal connection for a one phase of the compilation
